core/app/models/uffizzi_core/compose_file.rb

```diff
@@ -54,4 +53,12 @@ class UffizziCore::ComposeFile < UffizziCore::ApplicationRecord
       transitions from: [:valid_file], to: :invalid_file
     end
   end
+
+  private
+
+  def main_compose_file_uniqueness
+    return unless project.compose_files.main.exists?
+
+    errors.add(:compose_file, 'Main compose file already exist for this project')
```

Comment:
Let's not say `Main` compose file - it's a concept of a backend, I don't think we should show it to the user. Maybe just 'A compose file already...'